service/history/workflow/mutable_state_impl.go

```diff
@@ -1690,12 +1694,18 @@ func (ms *MutableStateImpl) addWorkflowExecutionStartedEventForContinueAsNew(
 
 	enums.SetDefaultContinueAsNewInitiator(&command.Initiator)
 
-	// Copy version stamp to new workflow only if:
-	// - command says to use compatible version
-	// - using versioning
 	var sourceVersionStamp *commonpb.WorkerVersionStamp
+	var inheritedBuildId string
 	if command.InheritBuildId {
-		sourceVersionStamp = worker_versioning.StampIfUsingVersioning(previousExecutionInfo.MostRecentWorkerVersionStamp)
+		inheritedBuildId = previousExecutionInfo.AssignedBuildId
```

Comment:
I guess the reason for not factoring out this logic out of two places is that it's going to be deleted?